src/metrics.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  defaultOptions,
+  FEATURE_IDENTIFIER_ATTRIBUTE,
+  GLOBAL_TARGET,
+} from './constants';
+import {
+  Configuration,
+  FeatureConfig,
+  Metrics,
+  MetricsApi,
+  Variation,
+  TargetData,
+  KeyValue,
+  MetricsData,
+  MetricsDataMetricsTypeEnum,
+} from './openapi';
+import { Options, Target } from './types';
+
+const log = defaultOptions.logger;
+
+interface AnalyticsEvent {
+  target: Target;
+  featureConfig: FeatureConfig;
+  variation: Variation;
+  count: number;
+}
+
+export interface MetricsProcessorInterface {
+  start(): void;
+  close(): void;
+  enqueue(
+    target: Target,
+    featureConfig: FeatureConfig,
+    variation: Variation,
+  ): void;
+}
+
+export const MetricsProcessor = (
+  environment: string,
+  cluster = '1',
+  conf: Configuration,
+  options: Options,
+): MetricsProcessorInterface => {
+  let data: Record<string, AnalyticsEvent>;
+  let syncInterval: NodeJS.Timeout;
+
+  const configuration = new Configuration({
+    ...conf,
+    basePath: options.eventsUrl,
+  });
+  const api = new MetricsApi(configuration);
+
+  const enqueue = (
+    target: Target,
+    featureConfig: FeatureConfig,
+    variation: Variation,
+  ): void => {
+    const event: AnalyticsEvent = {
+      target,
+      featureConfig,
+      variation,
+      count: 0,
+    };
+
+    const key = _formatKey(event);
+    const found = data[key];
+    if (found) {
+      found.count++;
+    } else {
+      event.count = 1;
+      data[key] = event;
+    }
+  };
+
+  const _formatKey = (event: AnalyticsEvent): string => {
+    const feature = event.featureConfig.feature,
+      variation = event.variation.identifier,
+      value = event.variation.value,
+      target = GLOBAL_TARGET;
+    return `${feature}/${variation}/${value}/${target}`;
+  };
+
+  const _summarize = (): Metrics | unknown => {
+    if (!data) {
+      log.debug('No metrics data!');
+      return;
+    }
+    const targetData: TargetData[] = [];
+    const metricsData: MetricsData[] = [];
+
+    for (const [, event] of Object.entries(data)) {
```

Comment:
You can simplify a little bit with `Object.values` since you don't need the key.
```suggestion
    for (const event of Object.values(data)) {
```

Reply:
This looks like a decent use-case for a `reduce`